### DISCUSSION TITLE
Native Steam Deck HTML Elements?

### DISCUSSION THREAD

**Discussion by heavyelement** [SteamDeckHomebrew/decky-loader#4] [Q&A] [answered]
Does anyone know what classes/elements need to be added to make items appear as native UI elements? Adding an `<input type="checkbox">` only shows a normal checkbox. How do we create pretty ranges, checkboxes, etc?

**Comment by TrainDoctor** [comment]
This should address this pretty well. 😄 https://deckbrew.xyz/plugin-dev/snippetsandsamples

**Comment by TrainDoctor** [comment, marked as answer]
This has been addressed via the usage of React and the [decky-frontend-lib](https://github.com/SteamDeckHomebrew/decky-frontend-lib) which is demonstrated in [decky-plugin-template](https://github.com/SteamDeckHomebrew/decky-plugin-template) and [decky-playground](https://github.com/SteamDeckHomebrew/decky-playground). These are the current best methods to have native UI elements in your plugin without having to use webpack in your own plugin to retrieve them. 👍🏻
